### PR TITLE
Usability-improovements-2

### PR DESCRIPTION
**Pull request by jdroenner**

### CODE REVIEW

**Comment by jdroenner** [comment]
[Python Usability Improvements 2](https://trello.com/c/1JiGY0cI/1354-python-usability-improvements-2)

File: geoengine/datasets.py [hunk 1]
```diff
@@ -425,13 +425,15 @@ def pandas_dtype_to_column_type(dtype: np.dtype) -> FeatureDataType:
 
 def upload_dataframe(
         df: gpd.GeoDataFrame,
-        name: str = "Upload from Python",
+        display_name: str = "Upload from Python",
+        name: Optional[str] = None,
```
Comment (ChristianBeilschmidt) [inline geoengine/datasets.py:429]: Don't you have to update at least one example?

Comment (ChristianBeilschmidt) [inline geoengine/datasets.py:429]: Maybe write a doc-comment, what happens when you leave it empty?

File: geoengine/workflow.py [hunk 2]
```diff
@@ -941,11 +942,14 @@ def __replace_http_with_ws(self, url: str) -> str:
         return f'{ws_prefix}{url_part}'
 
 
-def register_workflow(workflow: Dict[str, Any], timeout: int = 60) -> Workflow:
+def register_workflow(workflow: Union[Dict[str, Any], WorkflowBuilderOperator], timeout: int = 60) -> Workflow:
```
Comment (ChristianBeilschmidt) [inline geoengine/workflow.py:945]: Aren't there also examples that could leave the `to_workflow_dict` out?

**Review by ChristianBeilschmidt — CHANGES_REQUESTED** [review]

File: geoengine/datasets.py (hunk 1 above)
Comment (jdroenner) [reply geoengine/datasets.py:429]: I think the upload example does not use the "name". I can add one if you want.

File: geoengine/workflow.py (hunk 2 above)
Comment (jdroenner) [reply geoengine/workflow.py:945]: https://github.com/geo-engine/geoengine-python/pull/138/commits/ed5af4d86a4a72b02549a6da630e44379ffa4494

**Review by ChristianBeilschmidt — APPROVED** [review]

**Pull request merged**